src/notification-banner/notification-banner.tsx

```diff
@@ -40,23 +46,48 @@ export const NBComponent = ({
     // =============================================================================
     // EVENT HANDLERS
     // =============================================================================
-    const handleDismiss = () => {
+    const handleDismiss = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
         setVisible(false);
 
         if (dismissible && onDismiss) onDismiss();
     };
 
+    const handleBannerClick = (event: React.MouseEvent<HTMLInputElement>) => {
+        if (!onClick) return;
+        event.stopPropagation();
+        onClick();
+    };
+
     // =============================================================================
     // RENDER FUNCTIONS
     // =============================================================================
     if (!isVisible) return null;
 
+    const renderViewMore = () => (
+        <ViewMoreButton>
+            View more
+            <ArrowRightIcon />
+        </ViewMoreButton>
+    );
+
     return (
-        <Wrapper ref={forwardedRef} $sticky={sticky} {...otherProps}>
+        <Wrapper
+            ref={forwardedRef}
+            $sticky={sticky}
+            $clickable={!!onClick}
+            onClick={handleBannerClick}
+            {...otherProps}
+        >
             <Container id={formatId("container", id)}>
                 <TextContainer>
                     <Content data-testid={formatId("text-content", testId)}>
-                        {children}
+                        <ContentWrapper
+                            $collapsedHeight={collapsedHeight}
+                        >
+                            {children}
+                        </ContentWrapper>
+                        {!isNil(collapsedHeight) && renderViewMore()}
```

Comment:
the checks for `collapsedHeight` should be consistent - in styles, it's 
```
if (props.$collapsedHeight)
```
which doesn't allow for 0

checking for `!!collapsedHeight` should be sufficient here as setting 0 doesn't make sense